### PR TITLE
650 exec img comp

### PR DESCRIPTION
**Pull request by AzizPatel786**

### CODE REVIEW

**Comment by AzizPatel786** [comment]
https://github.com/user-attachments/assets/42c50de3-79bf-4626-b662-c7fc7ca17da8

File: client/src/components/generic/ExecImage/ExecImage.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const ExecImage: React.FC<ExecImageProps> = ({ src, alt, title, name }) => {
+  const [isHovered, setIsHovered] = useState(false)
+
+  const handleTouchStart = () => {
+    setIsHovered(true)
+  }
+
+  const handleTouchEnd = () => {
+    setIsHovered(false)
+  }
```
Comment (bcho892) [inline client/src/components/generic/ExecImage/ExecImage.tsx:21]: https://tailwindcss.com/docs/hover-focus-and-other-states

**Comment by AzizPatel786** [comment]
> Screen.Recording.2024-07-20.203547.mp4

mouse hover

**Comment by AzizPatel786** [comment]
https://github.com/user-attachments/assets/1eba6661-6566-4b43-a769-586662050171

touch hover

File: client/src/components/generic/ExecImage/ExecImage.css
Comment (bcho892) [inline client/src/components/generic/ExecImage/ExecImage.css:1]: We should use [tailwindcss](https://tailwindcss.com/) for this for consistency, you will find that all of these can be used with tailwind

File: client/src/components/generic/ExecImage/ExecImage.tsx (hunk 1 above)
Comment (bcho892) [reply client/src/components/generic/ExecImage/ExecImage.tsx:21]: Even though it is super easy, we should avoid javascript for styling whenever possible:

- CSS can take advantage of GPU acceleration more often compared to JS
- Javascript increases styling complexity and can lead to unnecessary re-renders

**Review by bcho892 — CHANGES_REQUESTED** [review]

File: client/src/components/generic/ExecImage/ExecImage.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from "react"
+import Image from "next/image"
+// import "./ExecImage.css"
```
Comment (bcho892) [inline client/src/components/generic/ExecImage/ExecImage.tsx:3]: nit: delete

File: client/src/components/generic/ExecImage/ExecImage.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <div className="group relative flex h-[163px] w-[163px] flex-shrink-0 overflow-hidden rounded-[4px]">
+      <Image
+        src={src}
+        alt={alt}
+        width={163}
+        height={163}
+        className="h-full w-full object-cover transition-opacity duration-300"
+      />
+      <div className="absolute left-0 top-0 h-full w-full bg-[rgba(40,61,135,0.8)] opacity-0 transition-opacity duration-300 group-hover:opacity-100"></div>
```
Comment (bcho892) [inline client/src/components/generic/ExecImage/ExecImage.tsx:22]: nit: 
```suggestion
      <div className="absolute left-0 top-0 h-full w-full bg-dark-blue-100 opacity-0 transition-opacity duration-300 group-hover:opacity-100"></div>
```

check out `client\tailwind.config.ts` for preset colours

**Review by bcho892 — APPROVED** [review]

File: client/src/components/generic/ExecImage/ExecImage.tsx (hunk 3 above)
Comment (AzizPatel786) [reply client/src/components/generic/ExecImage/ExecImage.tsx:22]: How do i make it 0.8 opacity

Comment (bcho892) [reply client/src/components/generic/ExecImage/ExecImage.tsx:22]: Suggest getting tailwind intellisense, there should be `opacity-80`

**Pull request merged**